test/libsolidity/SolidityExecutionFramework.h

```diff
@@ -65,6 +66,7 @@ class SolidityExecutionFramework: public solidity::test::ExecutionFramework
 
 	bytes multiSourceCompileContract(
 		std::map<std::string, std::string> const& _sources,
+		std::optional<std::string> const& _sourceName = std::nullopt,
```

Comment:
This is the name of the entry from `_sources` that can contain external sources?
```suggestion
		std::optional<std::string> const& _mainSourceName = std::nullopt,
```

And I think you should assert that `_sourceName` is in `_sources` (if not `nullopt`).